components/ui-api-layer/docs/proposals/exceed-resource-quota.md

```diff
@@ -0,0 +1,84 @@
+To calculate how many resources the ReplicaSet needs to progress, sum up the resource usage of all containers in the replica Pod. You must also calculate the difference from `.spec.hard` and `.status.used`. It gives you the total amount of available resources to use in the given Namespace.
+Loop through the ResourceQuota list and check if any ResourceQuota blocks the progress of the ReplicaSet by comparing the resources available in the Environment and the resources required to create the next replica.
+If there is not enough resources to create the next replica and the number of desired replicas is not reached, return a ResourceQuotaStatus with a flag set to `true`.
+
+>**NOTE** The Kubeless functions work with this solution because they use the Deployment and the ReplicaSet underneath.
+
+### Check the StatefulSets
+
+List StatefulSets and loop through them.
+For each StatefulSet that has a number of replicas lower than expected, implement the logic which checks if any ResourceQuota blocks the progress of the StatefulSet.
+To achieve that, implement the logic which calculates if the StatefulSet has enough memory to create the next replica.
+To calculate how many resources the StatefulSet needs to progress, sum up the resource usage of all containers in the replica Pod. You must also calculate the difference from `.spec.hard` and `.status.used`. It gives you the total number of resources available to use in the given Namespace.
+
+Loop through the ResourceQuota list and check if any ResourceQuota blocks the progress of the ReplicaSet by comparing the resources available in the Environment and the resources required to create the next replica.
+If there is not enough resources to create the next replica and the number of replicas is not reached, return a ResourceQuotaStatus with a flag set to `true`.
+
+### End of checking
+
+After all checks passed and the resources usage in the given Environment did not exceed the ResourceQuota, return the ResourceQuotaStatus with a flag set to `false`.
+
+[Here](https://github.com/dtaylor113/origin-web-console/blob/master/app/scripts/services/quota.js#L86) you can find the example implementation of the checking logic in UI.
+
+
+## Percentage usage of the ResourceQuota
+
+The percentage thresholds are stored in the ConfigMap. Each resource type has its own threshold.
+In the ConfigMap, there is also one default percentage threshold for all resource types which do not have a threshold set.
+
+In the future, the user will be able to configure thresholds stored in the ConfigMap.
+With a percentage approach, you can see the actual usage for every environment on the UI.
+
+The UI must mention the user when he exceeded the percentage threshold for a given resource.
+
+To avoid situation when the user exceeded the ResourceQuota without knowledge about it, the UI should notify him when any of the ReplicaSet in his namespace does not reach the desired number of Pods.
```

Comment:
`To avoid the situation when the user exceeded the ResourceQuota without realizing it, the UI should send a notification every time any ReplicaSet in the Namespace does not reach the desired number of Pods.`

Reply:
Done